packages/ember-glimmer/lib/syntax/mount.js

```diff
@@ -43,13 +44,20 @@ function dynamicEngineFor(vm, symbolTable) {
   @public
 */
 export function mountMacro(path, params, hash, builder) {
-  assert(
-    'You can only pass a single argument to the {{mount}} helper, e.g. {{mount "chat-engine"}}.',
-    params.length === 1 && hash === null
-  );
+  if (EMBER_ENGINES_MOUNT_PARAMS) {
+    assert(
+      'You can only pass a single positional argument to the {{mount}} helper, e.g. {{mount "chat-engine"}}.',
+      params.length === 1
+    );
+  } else {
+    assert(
+      'You can only pass a single argument to the {{mount}} helper, e.g. {{mount "chat-engine"}}.',
+      params.length === 1 && hash === null
+    );
+  }
 
   let definitionArgs = [params.slice(0, 1), null, null, null];
-  let args = [null, null, null, null];
+  let args = [params.slice(1), hash, null, null];
```

Comment:
Shouldn't the first arg continue being `null`? If `params.length === 1` then this `slice` seems a bit unnecessary.